src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorDiagnosticsEndpoint.cs

```diff
@@ -238,49 +238,64 @@ static bool CheckIfAttributeContainsNonMarkupNodes(RazorSyntaxNode attributeNode
             }
         }
 
-        private Diagnostic[] FilterCSharpDiagnostics(Diagnostic[] unmappedDiagnostics)
+        private Diagnostic[] FilterCSharpDiagnostics(Diagnostic[] unmappedDiagnostics, RazorCodeDocument codeDocument)
         {
             return unmappedDiagnostics.Where(d =>
-                !(DiagnosticsToIgnore.Contains(d.Code?.String) &&
-                  d.Severity != DiagnosticSeverity.Error))
-                .ToArray();
+                !ShouldFilterCSharpDiagnosticBasedOnErrorCode(d, codeDocument)).ToArray();
+        }
+
+        private bool ShouldFilterCSharpDiagnosticBasedOnErrorCode(Diagnostic diagnostic, RazorCodeDocument codeDocument)
+        {
+            if (!diagnostic.Code.HasValue)
+            {
+                return false;
+            }
+
+            return diagnostic.Code.Value.String switch
+            {
+                "CS1525" => IsInEmptyAttribute(diagnostic, codeDocument),
+                _ => CSharpDiagnosticsToIgnore.Contains(diagnostic.Code.Value.String) &&
+                        diagnostic.Severity != DiagnosticSeverity.Error,
+            };
+
+            bool IsInEmptyAttribute(Diagnostic diagnostic, RazorCodeDocument codeDocument)
+            {
+                if (TryGetOriginalDiagnosticRange(diagnostic.Range, diagnostic.Severity, codeDocument, out var originalRange) &&
```

Comment:
Hmm, how does this denote "in an empty attribute" though? I'm worried we'll be suppressing too much is all

Reply:
Because we have a C# diagnostic with a range relative to the generated C# generated document that isn't mappable to razor, there isn't a way for us to inspect the razor syntax tree and detect an empty attribute.

Admittedly this may be a bit heavy handed, but generally speaking unmapped ranges are usually indicative of non-razor relevant C# errors (like unused-usings for instance). Because we're specifically suppressing [`CS1525`](https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs1525) specifically when it's in an unmapped position, I don't think this should break the user document/experience.

Reply:
To ensure this function isn't used for any other purpose, I'll rename to `ShouldIgnoreCS1525`.

Reply:
Hmmm, can we see if Razor's diagnostic for an empty attribute also exists to make it a little less "blankety"?

Reply:
Sure, updated!